### PR TITLE
bundler tests and improve `Bun.build` return type

### PR DESCRIPTION
**Pull request by paperclover**

### CODE REVIEW

File: src/bun.js/javascript.zig [hunk 1]
```diff
@@ -2716,15 +2734,24 @@ pub const BuildError = struct {
         return ZigString.init(this.msg.data.text).toValue(ctx.ptr()).asRef();
```
Comment (Jarred-Sumner) [inline src/bun.js/javascript.zig]: we should mlake all of these `toValueGC`

it's probably the cause of the crash you're seeing

File: src/logger.zig [hunk 2]
```diff
@@ -700,7 +707,7 @@ pub const Log = struct {
         });
     }
 
-    pub fn toJS(this: Log, global: *JSC.JSGlobalObject, allocator: std.mem.Allocator, fmt: string) JSC.JSValue {
+    pub fn toJSError(this: Log, global: *JSC.JSGlobalObject, allocator: std.mem.Allocator, fmt: string) JSC.JSValue {
```
Comment (Jarred-Sumner) [inline src/logger.zig]:
```suggestion
    pub fn toJS(this: Log, global: *JSC.JSGlobalObject, allocator: std.mem.Allocator, fmt: string) JSC.JSValue {
```

the only usage where we don't want to throw it is in the bundler

let's keep it as `toJS` for consistentcy with the rest of the code

File: src/resolver/resolver.zig [hunk 3]
```diff
@@ -632,7 +632,7 @@ pub const Resolver = struct {
                 return r._resolveFramework(package, pair, preference, load_defines) catch |err| {
                     switch (err) {
                         error.ModuleNotFound => {
-                            Output.prettyErrorln("<r><red>ResolveError<r> can't find framework: <b>\"{s}\"<r>.\n\nMaybe it's not installed? Try running this:\n\n   <b>bun add -d {s}<r>\n   <b>bun bun --use {s}<r>", .{ package, package, package });
```
Comment (Jarred-Sumner) [inline src/resolver/resolver.zig]: nope we should keep these

File: test/bundler/acorn.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+minify_in_place() {
+  for file in $(find $1 -name '*.js'); do
+    echo "minifying $file"
+    bun build $file --target=node --minify --external '*' > $file.min.js
+    bun build $file.min.js > /dev/null
+    mv $file.min.js $file
+  done
+}
+minify_in_place acorn/src
+minify_in_place acorn-loose/src
```
Comment (Jarred-Sumner) [inline test/bundler/acorn.sh:43]: how long does this test take to run?

Comment (paperclover) [reply test/bundler/acorn.sh:43]: its mostly time to clone and install

8 seconds total. with repo cached its 3 seconds

**Pull request merged**